### ISSUE TITLE
Warning: hidden lifetime parameters in types are deprecated

### ISSUE THREAD

**Issue by lpil** [capnproto/capnproto-rust#206]
Hello! Thank you for this great library!

I'm getting the following warning from the generated code on Rust 1.48:

```
error: hidden lifetime parameters in types are deprecated
   --> src\bindata_capnp.rs:118:39
    |
118 |     pub fn set_name(&mut self, value: ::capnp::text::Reader)  {        
    |                                       ^^^^^^^^^^^^^^^^^^^^^- help: indicate the anonymous lifetime: `<'_>`
```

I set warnings as errors so this causes my build to break.

For now I'm copying the file into the source tree and editing it to fix the problem, but it would be great to have this warning fixed!

Thanks,
Louis

**Comment by dwrensha** [comment]
Hm... looks like you are running into the [`elided_lifetimes_in_paths`](https://github.com/rust-lang/rust/blob/e37f25aa3f356546ab851e394d5598fc575eabda/compiler/rustc_lint_defs/src/builtin.rs#L1541-L1574) lint. I'm kind of curious about how you are opting in to that? Do you explicitly have `#![warn(elided_lifetimes_in_paths)]` at the top of your crate?

I would suggest that you could put `#[allow(elided_lifetimes_in_paths)]` on the generated modules, but apparently that lint is only allowed to be configured at the crate level. Weird.

In any case, I agree that it would be good to generate code that did not trigger this lint.

**Comment by lpil** [comment]
You you got it! My company has a very comprehensive clippy configuration.

**Comment by lpil** [comment]
Hello! I dug more into this and it seems the lint that is causing the problem in our list is `rust_2018_idioms`.

**Comment by dwrensha** [comment]
Fixed in 4ee7dde25e7fe807a662906d8664ef16f5b029ba.

**Comment by lpil** [comment]
Thank you. Will there be another release with this fix soon?￼

**Comment by dwrensha** [comment]
The fix is included in [capnpc](https://crates.io/crates/capnpc) version 0.14.1.

**Comment by lpil** [comment]
Oh! Thank you, I got confused and thought a full compiler release was required. Thank you.
